### DISCUSSION TITLE
GSAP ScrollTrigger needed?

### DISCUSSION THREAD

**Discussion by TomSmedley** [darkroomengineering/lenis#140] [General]
Hey Studio Freight team!

Thank you for releasing Lenis open source! I've always avoided smooth scrolling libraries because of accessibility concerns or random bugs where it would scroll really slowly on a trackpad for example. This is the first library I've used in that respect that I feel like answers all concerns. I also want to congratulate you on V1 release!

I just wanted to ask about the new GSAP section in the documentation (https://github.com/studio-freight/lenis#gsap-scrolltrigger-integration). I have integrated this but I actually feel like ScrollTrigger animations feel much smoother without it. I was wondering what was the reason for suggesting this implementation with ScrollTrigger? 

Thanks,
Tom

**Comment by laurent-d** [comment]
Very interested about this, one of the benefit of your lib is the weight but if we add 40ko of gsap + scroll trigger module.. Perhaps a vanilla example would be nice :)

**Comment by rubek-joshi** [comment]
I too am facing this in my next 13 app. The docs mention this in particular:

> ```typescript
> lenis.on('scroll', ScrollTrigger.update)
> 
> gsap.ticker.add((time)=>{
>   lenis.raf(time * 1000)
> })
> ```

But as soon as I add this:
> ```typescript
> gsap.ticker.add((time)=>{
>   lenis.raf(time * 1000)
> })
> ```

my animations start becoming janky. When I remove it, my animation seems to work fine so I'm wondering if there'll be any kind of problem down the line due to the removal of this particular code snippet? (I'm thinking some kind of sync problems but I can't imagine what though)

**Comment by ledhieu** [comment]
Adding these worked for me
```javascript
gsap.ticker.add((time)=>{
  lenis.raf(time * 1000)
}, false, true)
```
----------------
**MY GUESS**

I think the original problem is caused by [forced reflow](https://yonatankra.com/layout-reflow/). When Lenis raf is added to GSAP ticker, GSAP does its animations and then Lenis does its scrolling. This measures the DOM after mutating it. In the Performance tab, style recalculation was done twice on every frame.

![image](https://github.com/user-attachments/assets/8d1c834c-5838-4cd8-aa81-a88e9b13eddd)
(each task took 11ms)

Adding (false, true) to the parameters (see [gsap.ticker()](https://gsap.com/docs/v3/GSAP/gsap.ticker()/)) prioritizes Lenis' operations, so now we can measure the DOM before mutating it.

![image](https://github.com/user-attachments/assets/933987b4-9cba-4bdb-970a-2bc9775256af)
(each task took 7ms)
